test/unit/expiration_process_test.lua

```diff
@@ -0,0 +1,104 @@
+local expirationd = require("expirationd")
+local fiber = require("fiber")
+local t = require("luatest")
+
+local helpers = require("test.helper")
+
+local g = t.group('expiration_process', {
+    {index_type = 'TREE', engine = 'vinyl'},
+    {index_type = 'TREE', engine = 'memtx'},
+    {index_type = 'HASH', engine = 'memtx'},
+})
+
+g.before_each({index_type = 'TREE'}, function(cg)
+    t.skip_if(cg.params.engine == 'vinyl' and helpers.vinyl_is_broken(),
+            'Blocked by https://github.com/tarantool/tarantool/issues/6448')
+    g.space = helpers.create_space_with_tree_index(cg.params.engine)
+    local space_archive = helpers.create_space('archived_tree', cg.params.engine)
+    space_archive:create_index("primary")
+    g.space_archive = space_archive
+end)
+
+g.before_each({index_type = 'HASH'}, function(cg)
+    g.space = helpers.create_space_with_hash_index(cg.params.engine)
+    local space_archive = helpers.create_space('archived_hash', cg.params.engine)
```

Comment:
Of course, I have less experience with tests, but I have not seen something so that the tests are always divided into two (preparatory part and asserts part), I saw a code where a function is called at the beginning of a test or a fixture is used, but I have not seen cases so that each test has a separate preparatory function or a fixture for a specific test.  Can you give an example from our repositories or others? Of course, I can do this, but I just want to know if this is a real case. In general, I was thinking of wrapping this preparatory code in a separate function with parameters, but since there are 5 lines, it would only confuse

Reply:
My suggestion is above
https://github.com/tarantool/expirationd/pull/99#discussion_r764191461